src/main/java/com/spencerwi/either/Either.java

```diff
@@ -41,6 +41,22 @@
 
     public abstract <R2> Either<L, R2> flatMapRight(Function<R, Either<L,R2>> transformRight);
 
+    public abstract L getLeftOrElse(L other);
```

Comment:
In thinking through this, I'm not sure that this method makes much sense (and the same goes for its overload and the corresponding `getRightOrElse` signatures).

I can't come up with a use-case where the surrounding code _does_ care about both the "left" and "right" possible values enough to use an `Either<L,R>` instead of an `Optional<L>` or an `Optional<R>`, but then _doesn't_ actually want to use the "right" possible value (since `getLeftOrElse` is basically just throwing away the right-side value when unwrapping) or _doesn't_ actually want to use the "left" possible value (since `getRightOrElse` is just throwing away the left-side value when unwrapping).

For `Optional`, the `orElse(T value)` method makes sense -- in absence of a value, you want to use some default. For `Either`, it doesn't make as much sense -- in absence of a right-side `R`, then you have a left-side `L`, and vice-versa;  if you don't have an `R` _and_ you don't _want_ the `L`, then you probably should've just used an `Optional<R>` (and, of course, vice-versa), and an `Either<L,R>` doesn't actually convey your intent (and is probably the incorrect choice).

That leaves us with only the `getLeftOrElseThrow`/`getRightOrElseThrow` methods, which only have utility in that they allow the user to _specify_ which `Throwable` value is thrown if wrong-side unwrap method is called. That has somewhat questionable utility, since it's isomorphic to:

```java
try {
    right.getLeft();
} catch (NoSuchElementException e) {
    throw new MySpecificException(e);
}
```

Perhaps that might be a bit verbose, but it does satisfy the same requirement using the code that exists today -- and `getLeft()` / `getRight()` are similar to Optional's `.get()` in that they should never be called without first checking `isLeft()` or `isRight()`.

Reply:
I'm fine. 
`getLeftOrElseThrow` / `getRightOrElseThrow`, with `Supplier`, is something I miss in my use case. Other methods were implemented due Vavr inspiration - I can't find real use case for them.